src/udp_bridge.h

```diff
@@ -0,0 +1,309 @@
+    License, v. 2.0. If a copy of the MPL was not distributed with this
+    file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+*/
+
+#pragma once
+
+#ifndef UDP_BRIDGE_H
+#define UDP_BRIDGE_H
+
+#define UDP_CONNECT_METHOD          "udp/connect"
+#define UDP_CONNECT_MULTI_METHOD    "udp/connectMulticast"
+#define UDP_CLOSE_METHOD            "udp/close"
+#define UDP_WRITE_METHOD            "udp/write"
+#define UDP_AWAIT_READ_METHOD       "udp/awaitRead"
+#define UDP_READ_METHOD             "udp/read"
+
+#include <api/Udp.h>
+
+#define DEFAULT_UDP_BUF_SIZE    4096
+
+
+struct BridgeUdpMeta {
+    MsgPack::str_t host;
+    uint16_t port;
+    uint16_t size;
+
+    BridgeUdpMeta() {
+        host = "";
+        port = 0;
+        size = 0;
+    }
+
+    MSGPACK_DEFINE(size, host, port); // -> [code, traceback]
+};
+
+template<size_t BufferSize=DEFAULT_UDP_BUF_SIZE>
+class BridgeUDP final: public UDP {
+
+    BridgeClass* bridge;
+    uint32_t connection_id{};
+    uint32_t read_timeout = 1;
+    RingBufferN<BufferSize> temp_buffer;
+    struct k_mutex udp_mutex{};
+    bool _connected = false;
+
+    uint16_t _port{}; // local port to listen on
+
+    // Outbound packets target
+    String _targetHost{};
+    uint16_t _targetPort{};
+
+    // Inbound packet info
+    IPAddress _remoteIP{}; // remote IP address for the incoming packet whilst it's being processed
+    uint16_t _remotePort{}; // remote port for the incoming packet whilst it's being processed
+    uint16_t _remaining{}; // remaining bytes of incoming packet yet to be processed
+    BridgeUdpMeta packet_meta{};
+
+public:
+
+    explicit BridgeUDP(BridgeClass& bridge): bridge(&bridge) {}
+
+    uint8_t begin(uint16_t port) override {
+
+        if (!init()) {
+            return 0;
+        }
+
+        k_mutex_lock(&udp_mutex, K_FOREVER);
+
+        String hostname = "0.0.0.0";
+        const bool ok = _connected || bridge->call(UDP_CONNECT_METHOD, hostname, port).result(connection_id);
+        _connected = ok;
+        if (_connected) _port = port;
+
+        k_mutex_unlock(&udp_mutex);
+
+        return ok? 1 : 0;
+    }
+
+    void setTimeout(const uint32_t ms) {
+        k_mutex_lock(&udp_mutex, K_FOREVER);
+        read_timeout = ms;
+        k_mutex_unlock(&udp_mutex);
+    }
+
+    uint8_t beginMulticast(IPAddress ip, uint16_t port) override {
+
+        if (!init()) {
+            return 0;
+        }
+
+        k_mutex_lock(&udp_mutex, K_FOREVER);
+
+        String hostname = ip.toString();
+        const bool ok = _connected || bridge->call(UDP_CONNECT_MULTI_METHOD, hostname, port).result(connection_id);
+        _connected = ok;
+        if (_connected) _port = port;
+
+        k_mutex_unlock(&udp_mutex);
+
+        return ok? 1 : 0;
+    }
+
+    void stop() override {
+        k_mutex_lock(&udp_mutex, K_FOREVER);
+
+        if (_connected) {
+            String msg;
+            _connected = !bridge->call(UDP_CLOSE_METHOD, connection_id).result(msg);
```

Comment:
The `udp/close` will [always close the connection](https://github.com/arduino/arduino-router/blob/3a4b4617d3dfa6075f515d304fd97472a98614f4/network-api/network-api.go#L524-L529) and drop the `connection_id` on the Linux side.
```suggestion
            bridge->call(UDP_CLOSE_METHOD, connection_id).result(msg);
            _connected = false;
```